Jellyfin.Server.Implementations/Devices/DeviceManager.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            await using var dbContext = _dbProvider.CreateContext();
+            await dbContext.Database
+                .ExecuteSqlInterpolatedAsync($"UPDATE [DeviceOptions] SET [CustomName] = {options.CustomName}")
+                .ConfigureAwait(false);
+
+            DeviceOptionsUpdated?.Invoke(this, new GenericEventArgs<Tuple<string, DeviceOptions>>(new Tuple<string, DeviceOptions>(deviceId, options)));
+        }
+
+        /// <inheritdoc />
+        public async Task<Device> CreateDevice(Device device)
+        {
+            await using var dbContext = _dbProvider.CreateContext();
+
+            dbContext.Devices.Add(device);
+
+            await dbContext.SaveChangesAsync().ConfigureAwait(false);
+            return device;
+        }
+
+        /// <inheritdoc />
+        public async Task<DeviceOptions?> GetDeviceOptions(string deviceId)
+        {
+            await using var dbContext = _dbProvider.CreateContext();
+            return await dbContext.DeviceOptions
+                .AsQueryable()
+                .FirstOrDefaultAsync(d => d.DeviceId == deviceId)
+                .ConfigureAwait(false);
+        }
+
+        /// <inheritdoc />
+        public ClientCapabilities GetCapabilities(string deviceId)
+        {
+            return _capabilitiesMap.TryGetValue(deviceId, out ClientCapabilities? result)
+                ? result
+                : new ClientCapabilities();
+        }
+
+        /// <inheritdoc />
+        public async Task<DeviceInfo?> GetDevice(string id)
+        {
+            await using var dbContext = _dbProvider.CreateContext();
+            var device = await dbContext.Devices
+                .AsQueryable()
+                .Where(d => d.DeviceId == id)
+                .OrderByDescending(d => d.DateLastActivity)
+                .Include(d => d.User)
+                .FirstOrDefaultAsync()
+                .ConfigureAwait(false);
+
+            var deviceInfo = device == null ? null : ToDeviceInfo(device);
+
+            return deviceInfo;
+        }
+
+        /// <inheritdoc />
+        public async Task<QueryResult<Device>> GetDevices(DeviceQuery query)
+        {
+            await using var dbContext = _dbProvider.CreateContext();
+
+            var devices = dbContext.Devices.AsQueryable();
+
+            if (query.UserId.HasValue)
+            {
+                devices = devices.Where(device => device.UserId == query.UserId.Value);
+            }
+
+            if (query.DeviceId != null)
+            {
+                devices = devices.Where(device => device.DeviceId == query.DeviceId);
+            }
+
+            if (query.AccessToken != null)
+            {
+                devices = devices.Where(device => device.AccessToken == query.AccessToken);
+            }
+
+            if (query.Skip.HasValue)
+            {
+                devices = devices.Skip(query.Skip.Value);
+            }
+
+            var count = await devices.CountAsync().ConfigureAwait(false);
```

Comment:
I think `Count` should be calculated before skipping any records

Reply:
You're right, fixed